diesel_tests/Cargo.toml

```diff
@@ -9,15 +9,15 @@ autotests = false
 [build-dependencies]
 diesel = { path = "../diesel", default-features = false }
 diesel_migrations = { version = "1.4.0" }
-dotenv = ">=0.8, <0.11"
+dotenv = ">=0.8, <=0.15"
 
 [dependencies]
 assert_matches = "1.0.1"
 chrono = { version = "0.4" }
 diesel = { path = "../diesel", default-features = false, features = ["quickcheck", "chrono", "uuidv07", "serde_json", "network-address", "numeric", "with-deprecated"] }
 diesel_migrations = { version = "1.4.0" }
 diesel_infer_schema = { version = "1.4.0" }
-dotenv = ">=0.8, <0.11"
+dotenv = ">=0.8, <=0.15"
```

Comment:
This breaks the tests even back on older compiler versions with the following error:
```
error[E0659]: `error_chain` is ambiguous (derive helper attribute vs any other name)
  --> /home/weiznich/.cargo/registry/src/github.com-1ecc6299db9ec823/dotenv-0.10.1/src/lib.rs:23:40
   |
23 | #[cfg_attr(not(feature = "backtrace"), error_chain(backtrace = "false"))]
   |                                        ^^^^^^^^^^^ ambiguous name
   |
note: `error_chain` could refer to the derive helper attribute defined here
  --> /home/weiznich/.cargo/registry/src/github.com-1ecc6299db9ec823/dotenv-0.10.1/src/lib.rs:22:17
   |
22 | #[derive(Debug, error_chain)]
   |                 ^^^^^^^^^^^
note: `error_chain` could also refer to the derive macro imported here
  --> /home/weiznich/.cargo/registry/src/github.com-1ecc6299db9ec823/dotenv-0.10.1/src/lib.rs:10:1
   |
10 | #[macro_use]
   | ^^^^^^^^^^^^

error: aborting due to previous error
```

Reply:
Okay, so no more allowing 0.10

Reply:
Disallowing `0.10' here does not fix that, because `diesel-infer-schema` pulls in the '0.10' version.

Reply:
Luckily, it seems dotenv support in diesel_infer_schema is optional. I'll try just turning it off.

Reply:
That will cause CI to be fixed but break people's ability to run tests locally. We probably just need to get rid of `diesel_infer_schema` from our test suite

Reply:
That's probably better. All of the deprecated stuff is removed in the next release, anyway.

Reply:
That's also a pretty invasive change, though. Somebody who knows more about diesel should do it in a separate PR, and this one's blocked on it.

Reply:
I think it's fine to just leave the old dotenv version here, because that one continues to work. #2197 removes the dependency on `diesel_infer_schema`.